platform/android/java/lib/src/org/godotengine/godot/vulkan/VkRenderer.kt

```diff
@@ -48,52 +53,56 @@ import android.view.Surface
  */
 internal class VkRenderer {
 
+	private var pluginRegistry: GodotPluginRegistry = GodotPluginRegistry.getPluginRegistry()
+
 	/**
 	 * Called when the surface is created and signals the beginning of rendering.
 	 */
 	fun onVkSurfaceCreated(surface: Surface) {
-		nativeOnVkSurfaceCreated(surface)
+		GodotLib.surfacecreated(surface)
+		for (plugin in pluginRegistry.getAllPlugins()) {
+			plugin.onVkSurfaceCreated(surface)
+		}
 	}
 
 	/**
 	 * Called after the surface is created and whenever its size changes.
 	 */
 	fun onVkSurfaceChanged(surface: Surface, width: Int, height: Int) {
-		nativeOnVkSurfaceChanged(surface, width, height)
+		GodotLib.resize(width, height)
+		GodotLib.surfacecreated(surface)
+		for (plugin in pluginRegistry.getAllPlugins()) {
+			plugin.onVkSurfaceChanged(surface, width, height)
+		}
 	}
 
 	/**
 	 * Called to draw the current frame.
 	 */
 	fun onVkDrawFrame() {
-		nativeOnVkDrawFrame()
+		GodotLib.step()
+		for (plugin in pluginRegistry.getAllPlugins()) {
+			plugin.onVkDrawFrame()
+		}
 	}
 
 	/**
 	 * Called when the rendering thread is resumed.
 	 */
 	fun onVkResume() {
-		nativeOnVkResume()
+		GodotLib.onRendererResumed()
 	}
 
 	/**
 	 * Called when the rendering thread is paused.
 	 */
 	fun onVkPause() {
-		nativeOnVkPause()
+		GodotLib.onRendererPaused()
 	}
 
 	/**
 	 * Called when the rendering thread is destroyed and used as signal to tear down the Vulkan logic.
 	 */
 	fun onVkDestroy() {
-		nativeOnVkDestroy()
 	}
-
-	private external fun nativeOnVkSurfaceCreated(surface: Surface)
-	private external fun nativeOnVkSurfaceChanged(surface: Surface, width: Int, height: Int)
-	private external fun nativeOnVkResume()
-	private external fun nativeOnVkDrawFrame()
-	private external fun nativeOnVkPause()
-	private external fun nativeOnVkDestroy()
```

Comment:
Since you removed the `native*` callbacks, how are they propagated to the native layer?

Reply:
They are propagated through calls to `GodotLib` from `VkRenderer`, like `GodotRenderer` does for OpenGL.